server/lib/host-metrics.js

```diff
@@ -0,0 +1,218 @@
+import config from 'config';
+import moment from 'moment';
+
+import { sequelize } from '../models';
+
+import { getFxRate } from './currency';
+import { parseToBoolean } from './utils';
+
+async function computeTotal(results, currency) {
+  let total = 0;
+
+  // For sanity reasons, we handle conversion in case there is any currency mismatch
+  for (const result of results) {
+    const value = result['_amount'];
+    const fxRate = await getFxRate(result['_currency'], currency);
+    total += Math.round(value * fxRate);
+  }
+
+  return total;
+}
+
+function computeDates(startDate, endDate) {
+  startDate = startDate ? moment(startDate) : moment().utc().startOf('month');
+  endDate = endDate ? moment(endDate) : moment(startDate).utc().endOf('month');
+
+  return { startDate: startDate.toISOString(), endDate: endDate.toISOString() };
+}
+
+export async function getPlatformTips(host, { startDate, endDate } = {}) {
+  const results = await sequelize.query(
+    `SELECT
+  SUM(
+    CASE
+      WHEN t2."data"->>'hostToPlatformFxRate' IS NOT NULL THEN
+        t2."amountInHostCurrency"::numeric / (t2."data"->>'hostToPlatformFxRate')::numeric
+      ELSE
+        t2."amountInHostCurrency"
+    END
+  ) as "_amount",
+  (
+    CASE
+      WHEN t2."data"->>'hostToPlatformFxRate' IS NOT NULL THEN
+        h."currency"
+      ELSE
+        t2."hostCurrency"
+    END
+   ) as "_currency"
+FROM "Transactions" as t1
+INNER JOIN "Transactions" as t2
+ON t1."TransactionGroup" = t2."TransactionGroup"
+INNER JOIN "Collectives" as h
+ON t1."HostCollectiveId" = h."id"
+WHERE t1."HostCollectiveId" = :HostCollectiveId
+AND t1."createdAt" >= :startDate AND t1."createdAt" <= :endDate
+AND (t1."kind" IS NULL OR t1."kind" NOT IN ('PLATFORM_TIP'))
```

Comment:
I realize we haven't added any index on `kind`. We could maybe look into that when we'll want to optimize these queries.